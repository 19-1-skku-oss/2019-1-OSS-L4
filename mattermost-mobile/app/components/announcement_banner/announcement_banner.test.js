// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import Preferences from 'mattermost-redux/constants/preferences';

import AnnouncementBanner from './announcement_banner.js';

jest.useFakeTimers();

describe('AnnouncementBanner', () => {
    const baseProps = {
        bannerColor: '#ddd',
        bannerDismissed: false,
        bannerEnabled: true,
        bannerText: 'Banner Text',
        bannerTextColor: '#fff',
        navigator: {},
        theme: Preferences.THEMES.default,
    };

    test('should match snapshot', () => {
        const wrapper = shallow(
            <AnnouncementBanner {...baseProps}/>
        );

        expect(wrapper.getElement()).toMatchSnapshot();

        wrapper.setProps({bannerEnabled: false});
        expect(wrapper.getElement()).toMatchSnapshot();
    });
});
