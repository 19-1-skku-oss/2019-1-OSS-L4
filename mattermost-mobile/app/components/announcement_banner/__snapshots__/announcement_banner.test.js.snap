// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnnouncementBanner should match snapshot 1`] = `
<AnimatedComponent
  style={
    Array [
      Object {
        "overflow": "hidden",
        "paddingHorizontal": 10,
        "position": "absolute",
        "top": 0,
        "width": "100%",
      },
      Object {
        "backgroundColor": "#ddd",
        "height": 0,
      },
    ]
  }
>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <Text
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Array [
          Object {
            "flex": 1,
            "fontSize": 14,
            "marginRight": 5,
          },
          Object {
            "color": "#fff",
          },
        ]
      }
    >
      <RemoveMarkdown
        value="Banner Text"
      />
    </Text>
    <Icon
      allowFontScaling={false}
      color="#fff"
      name="info"
      size={16}
    />
  </TouchableOpacity>
</AnimatedComponent>
`;

exports[`AnnouncementBanner should match snapshot 2`] = `null`;
